### DISCUSSION TITLE
не пушит

### DISCUSSION THREAD

**Discussion by LessiaSa** [community/community#122366] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

не хочет работать push через командную строку. На GitHub Desktop push проходит моментально, а когда пытаюсь запушить через командную строку - висит сутками - и ни ответа ни привета. Ключ SSH поменяла, name и  email соответствуют, разрешение в настройках на чтение и запись, антивирус не блочит(на всякий случай еще и добавила в исключения сайт GitHub), GitHub отслеживает и добавила в commit. Уже не знаю какие заклинания и танцы с бубнами применять - памагити! Что еще сделать, чтоб push наконец пошел?

**Comment by Omega403** [comment, marked as answer]
Hi, I'm answering you in English because I'm not very familiar with writing in Russian. 
So, from what I understand, "`push`" via CMD doesn't work , while it goes with GitHub Desktop... have you tried checking the remote URL, checking that you are actually using SSH and possibly updating it? 

In the sense: 
With the CMD navigate to your git repository and then launch the 
`git remote -v` 
command which gives you your current URL. 

At this point check if you are actually using the SSH protocol and if not, launch the command 
`git remote set-url origin git@github.com:username/repo.git`
to set and use SSH (obviously replace name/ repo.git with your username and repository name). 

Finally, check with the command 
`ssh -T git@github.com`
that SSH is correctly configured and see if it gives you a confirmation message to see if everything is OK.

When you've done all this, mybe you should be able to push the changes from the command line

**Reply by LessiaSa** [reply]
Спасибо огромное, что откликнулись!
Протокол ssh да, использую, но такое впечатление, что с GitHub совсем не хочет связываться:
http://icecream.me/453bde77fdf463961e14fb257b9210e2

**Reply by Omega403** [reply]
Don't worry, no problem. 
Uhmm Ok... I would also tell you to check that Firewall and possibly Antivirus do not block git traffic, in case set github as an exception in both.
 
Then, I don't know but if you use proxies to connect to the network, check that they are set correctly, you can do this with the command 
`set HTTP_PROXY=http://proxy.com:port`
obviously replace _proxy.com_ and _port_ with your proxy settings. 

Also check that the Git configuration on your PC is correct. Then verify that the email address and username configured in Git match those of your Github account .
You can control these settings by running the following commands:
```
git config --global user.email
git config --global user.name
```
If the values ​​do not match you can change the credentials with:
```
git config --global user.email "...@email.com"
git config --global user.name "your-username"
```
If you want, you can try to disable git credential caching with the command:

`git config --global --unset credential.helper
`

then try the _push_ again from the CMD and manually enter the credentials when prompted. 
! ) This is because there could be an error if you have not memorized the credentials correctly and therefore you could be blocked during the _push_ phase
In fact I would invite you to check where the credentials are stored you can do it with 
`git config --global credential.helper
`
 and if you have "_manager_" or "_cache_" as the answer, the credentials are probably stored in the _**helper**_ and therefore it's fine

If you are sure you have checked everything, try relaunching the _push_ but with the _-v_ parameter to see if it gives you more details or errors

`git push -v
`
Or you can view the logs to see if you can find any errors

```
git log
```
Try these tests and make sure there are no problems with your connection

**Reply by LessiaSa** [reply]
спасибо огромное за отклик!
вот что получается
http://icecream.me/0766755342b13c8922bb790d5c59b5c8

**Reply by LessiaSa** [reply]
а здесь молчит, сейчас буду еще и с этим разбираться
http://icecream.me/9b36651692634a65ce58bb16e3a12cec

**Reply by LessiaSa** [reply]
ssh
http://icecream.me/701a67aa9b638075b620a8c59095fa29

**Reply by LessiaSa** [reply]
global list
http://icecream.me/01e667755a7588bf459febab9950634a

**Reply by LessiaSa** [reply]
все, решила переустановкой git на компьютере
Спасибо вам огромное!!!!

**Reply by Omega403** [reply]
I'm glad you solved it ^-^, actually reinstalling/rebooting is sometimes the best solution. I also took a look at the screens you sent and it didn't seem like there was anything wrong... so maybe it was just a communication problem. I'm glad I tried to help and I wish you good luck :) .

**Comment by ebndev** [comment]
Hi @LessiaSa, thanks for participating!

I'm glad you were able to get your question answered!

Please note: We take our [Code of Conduct](https://github.com/community/community/blob/920a928ffe3ae4863b64a4ea26917046e11f5c1a/CODE_OF_CONDUCT.md) very seriously and want to help ensure that everyone has a good experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. We’ll be locking any posts in languages other than English for now, including this one.
